src/main/java/com/validationtutorial/task1/UserRequest.java

```diff
@@ -0,0 +1,40 @@
+package com.validationtutorial.task1;
+
+import jakarta.validation.constraints.AssertTrue;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Size;
+
+public class UserRequest {
+    @NotBlank(message = "ユーザー名を入力してください")
+    @Size(min = 3, max = 20, message = "ユーザー名は3文字以上20文字以下である必要があります")
```

Comment:
ここのメッセージですがminとmaxで動的に生成するようにしたいです。

Reply:
動的に生成する点ですが、下記のような実装のイメージであっていますでしょうか。
```suggestion
    @Size(min = 3, max = 20, message = "ユーザー名は{min}文字以上{max}文字以下である必要があります")
```

Reply:
そんなイメージです！
---
あとデフォルトのバリデーションメッセージを使うパターンと今回のようにカスタムしたメッセージを作るのは課題にしてみるとよいですね。

ValidationMessage.propertiesを使う練習をしてもいいですし。

Reply:
承知しました！

> あとデフォルトのバリデーションメッセージを使うパターンと今回のようにカスタムしたメッセージを作るのは課題にしてみるとよいですね。
>ValidationMessage.propertiesを使う練習をしてもいいですし。

上記については、課題2の方で取り扱います。

Reply:
こちらで修正しました。
https://github.com/reytech-co-jp/validation-tutorial/pull/1/commits/66dcb079831193defc6295189e6213801a3b5935